Frameworks/CoreText/CTFont.mm

```diff
@@ -101,25 +110,74 @@
 const CFStringRef kCTFontFeatureSelectorSettingKey = static_cast<CFStringRef>(@"kCTFontFeatureSelectorSettingKey");
 
 /**
+ All these values are fixed number in IOS no matter on iphone* or ipad*.
+ */
+const CGFloat kCTFontLabelFontSize = 17.0f;
+const CGFloat kCTFontButtonFontSize = 14.0f;
+const CGFloat kCTFontSmallSystemFontSize = 12.0f;
+const CGFloat kCTFontSystemFontSize = 14.0f;
+
+using namespace Microsoft::WRL;
+
+struct __CTFont {
+    CFRuntimeBase _base;
+    ComPtr<IDWriteFont> _dwriteFont;
+    CGFloat _size;
+};
+
+static Boolean __CTFontEqual(CFTypeRef cf1, CFTypeRef cf2) {
+    struct __CTFont* font1 = (struct __CTFont*)cf1;
+    struct __CTFont* font2 = (struct __CTFont*)cf2;
+    return (font1->_size == font2->_size) && (static_cast<IDWriteFont3*>(font1->_dwriteFont.Get())->Equals(font2->_dwriteFont.Get()));
+}
+
+static CFHashCode __CTFontHash(CFTypeRef cf) {
+    CTFontRef font = (CTFontRef)cf;
+    return CFHash(CFAutorelease(CTFontCopyFullName(font))) * CTFontGetSize(font);
+}
+
+static CFStringRef __CTFontCopyDescription(CFTypeRef cf) {
+    CTFontRef font = (CTFontRef)cf;
+    return CFStringCreateWithFormat(kCFAllocatorSystemDefault,
+                                    NULL,
+                                    CFSTR("<CTFont %p>{font-family: %hs; font-size: %.2f}"),
+                                    cf,
+                                    [[static_cast<NSString*>(CTFontCopyFamilyName(font)) autorelease] UTF8String],
+                                    CTFontGetSize(font));
+}
+
+static CFTypeID __kCTFontTypeID = _kCFRuntimeNotATypeID;
+
+static const CFRuntimeClass __CTFontClass = { 0,
+                                              "CTFont",
+                                              NULL, // init
+                                              NULL, // copy
+                                              NULL, // deallocate (ComPtr takes care of RAII)
```

Comment:
>NULL [](start = 46, length = 4)

However, we do need a deallocator here. `~__CTFont()` only gets called if this `__CTFont` is `delete`d.
You'll probably have to implement a deallocator that explicitly placement-deletes the ComPtr and any other non-trivial types:

```
void __CTFontDeallocate(...) {
    thing->_dwriteFont->~ComPtr();
    // ...
}
```